src/awkward/highlevel.py

```diff
@@ -1531,7 +1535,19 @@ def __reduce_ex__(self, protocol: int) -> tuple:
             behavior = None
         else:
             behavior = self._behavior
-        return object.__new__, (Array,), (form.to_dict(), length, container, behavior)
+
+        if self._attrs is None:
+            attrs = self._attrs
+        else:
+            attrs = without_transient_attrs(self._attrs)
+
+        return unpickle_array_schema_1, (
+            form.to_dict(),
+            length,
+            container,
+            behavior,
+            attrs,
+        )
```

Comment:
@jpivarski this is something I'd like you to pay attention to — it means that newer Awkward Array pickles won't unpickle in older Awkward Array versions (even in the 2.x series). 

However, this PR will have to break unpickling in one form or another; it introduces a new state item that older Awkward versions won't be able to handle.

The intention of writing new, named functions is to encode the pickle scheme as a function name, rather than a version number alongside the pickle state. That way, older pickles are easily distinguished from newer ones in future versions of Awkward.

Reply:
Breaking forward-compatibility in pickles is bound to happen, I agree. Is this PR going to be in 2.5.0? That would be a good boundary at which to introduce such breakage. If someone is having trouble with pickled arrays, we can tell them that this is the boundary to avoid.